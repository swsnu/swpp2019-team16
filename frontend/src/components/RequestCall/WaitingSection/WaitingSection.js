import React from 'react';
import { makeStyles } from '@material-ui/styles';

const useStyles = makeStyles({
  root: {},
});

function WaitingSection() {
  const styles = useStyles();
  return <div className={styles.root}>Waiting for group to be matched...</div>;
}

export default WaitingSection;
