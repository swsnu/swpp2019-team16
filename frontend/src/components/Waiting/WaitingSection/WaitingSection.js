import React, { useState, useEffect } from 'react';
import quoteGenerator from 'inspirational-quotes';
import { makeStyles } from '@material-ui/styles';
import CircularProgress from '@material-ui/core/CircularProgress';
import { Typography } from '@material-ui/core';

const useStyles = makeStyles({
  root: {
    height: '600px',
    marginTop: '',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
  },
  title: {
    marginBottom: '3rem',
  },
  progress: {
    marginBottom: '3rem',
  },
  quote: {
    height: '350px',
    maxWidth: '500px',
    textAlign: 'center',
  },
});

function WaitingSection() {
  const styles = useStyles();
  const [quote, setQuote] = useState(null);

  useEffect(() => {
    const taskId = setInterval(() => {
      setQuote(quoteGenerator.getRandomQuote());
    }, 8000);
    return () => clearInterval(taskId);
  }, []);

  return (
    <div className={styles.root}>
      <div className={styles.title}>
        <Typography variant={'h4'}>We are making carpool group</Typography>
      </div>
      <div className={styles.progress}>
        <CircularProgress />
      </div>
      <div className={styles.quote}>
        <Typography variant={'body1'}>{quote}</Typography>
      </div>
    </div>
  );
}

export default WaitingSection;
