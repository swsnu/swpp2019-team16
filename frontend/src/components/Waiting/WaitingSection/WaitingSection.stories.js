import React from 'react';
import { storiesOf } from '@storybook/react';

import WaitingSection from './WaitingSection';
import { ThemeProvider } from '@material-ui/styles';
import theme from '../../../lib/styles/theme';
import { CssBaseline } from '@material-ui/core';
import { BrowserRouter } from 'react-router-dom';

storiesOf('Waiting/WaitingSection', module)
  .addDecorator(story => (
    <BrowserRouter>
      <ThemeProvider theme={theme}>
        <CssBaseline />
        {story()}
      </ThemeProvider>
    </BrowserRouter>
  ))
  .add('default', () => <WaitingSection />);
