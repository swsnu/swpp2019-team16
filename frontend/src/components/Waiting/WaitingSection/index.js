import WaitingSection from './WaitingSection';

export default WaitingSection;
