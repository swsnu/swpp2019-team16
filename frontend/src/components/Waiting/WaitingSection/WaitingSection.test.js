import React from 'react';
import { render } from '@testing-library/react';
import WaitingSection from './WaitingSection';

describe('<WaitingSection />', () => {
  it('SHOULD match with snapshot', async () => {
    const { container } = render(<WaitingSection />);

    expect(container).toMatchSnapshot();
  });
});
