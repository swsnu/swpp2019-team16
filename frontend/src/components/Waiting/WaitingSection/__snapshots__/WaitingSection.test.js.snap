// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WaitingSection /> SHOULD match with snapshot 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="makeStyles-title-2"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4"
      >
        We are making carpool group
      </h4>
    </div>
    <div
      class="makeStyles-progress-3"
    >
      <div
        class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </div>
    </div>
    <div
      class="makeStyles-quote-4"
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
      />
    </div>
  </div>
</div>
`;
